app/routes/public/role-invites.js

```diff
@@ -1,21 +1,25 @@
 import Route from '@ember/routing/route';
 
 export default Route.extend({
-  beforeModel(transition) {
+  async beforeModel(transition) {
     let payload = {
       data: {
-        token: transition.queryParams.token
+        token: transition.to.queryParams.token
       }
     };
-    this.loader
+    await this.loader
       .post('/role_invites/user', payload)
       .then(user => {
-        if (this.get('session.isAuthenticated')) {
-          if (this.get('authManager.currentUser.email') === user.email) {
+        if (this.session.isAuthenticated) {
+          if (this.authManager.currentUser.email === user.email) {
             this.loader
```

Comment:
@niranjan94 adding an `await` here threw a build error so removed it, will it be any problem for now?

Reply:
@uds5501 you cannot use `await` without using `async` perhaps that's why you had gotten the error.

Reply:
But the implementation is still wrong. When using async-await, don't use then-catch. 

Please [read up more on async-await](https://javascript.info/async-await).